### PR TITLE
Aryan - Onboarding Challenge

### PR DESCRIPTION
**Pull request by AesthicEthics**
# Purpose
Onboarding Challenge Review PR

### CODE REVIEW

File: drivers/source/serial_io.c [hunk 1]
```diff
@@ -64,7 +66,20 @@ obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
 
     /* USER CODE BEGIN */
     // Print text to the serial port using sciSendBytes. Use the mutex to protect the SCI module.
+    // take mutex
+    if (sciLinMutex != NULL){
```
Comment (dgobalak) [inline drivers/source/serial_io.c]: This check is done on line 64 so u don't need to do it again.

File: drivers/source/serial_io.c [hunk 2]
```diff
@@ -64,7 +66,20 @@ obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
...
+        if (xSemaphoreTake( sciLinMutex, ( TickType_t ) UART_MUTEX_BLOCK_TIME == pdTRUE)){
```
Comment (dgobalak) [inline drivers/source/serial_io.c]: Use `mutex` so we can support printing using either the SCI1 or SCI2/sciLin ports (depending on what UART_PRINT_REG  is defined as).

**Review by dgobalak — CHANGES_REQUESTED** [review]
Looks good. Just a few comments

File: drivers/source/serial_io.c (hunk 2 above)
Comment (Navtajh04) [inline drivers/source/serial_io.c]: this should be `if (xSemaphoreTake( sciLinMutex, ( TickType_t ) UART_MUTEX_BLOCK_TIME) == pdTRUE)` ie you're trying to compare the return value of xSemaphoreTake to see if it's successful and not UART_MUTEX_BLOCK_TIME

**Review by dgobalak — COMMENTED** [review]
lgtm. I'll close this PR now

**Pull request closed without merging**